### PR TITLE
Some fatal mistakes that needed to fix (Match Eps and Image Scale Appropriately)

### PR DESCRIPTION
**Pull request by 1105042987**
1. After `transforms.ToTensor()`, the scale of input image will change to [0,1], so that the variable `vgg_mean` in train_generator.py is too large for normalization. At the same time,  the variable `e_lim` is also 10 times larger than image.  we  need to transer it to 10/255. (That's also the reason why your fooling rate reached more than 99%)

2.  In normalization, we need to do `{}/std`, so that we need to change the order of normalization. That is, change `Norm(img)+noise` to `Norm(img+noise)`.

3. When training, `optimizer.zero_grad()` function is called at the begining of one batch. In the last version, the grad is canceled before updating the param in Generator.

### CODE REVIEW

**Comment by Gokkulnath** [comment]
Hi @1105042987
 
You are correct. Scale of Image and Epsilon should be the same. I had it fixed in my local version of the code, which i currently don't have access to.
 P.S This repo is not update to date with the changes that was made after the commit.  If i recollect correctly, i had to make changes with adoption of  VirtualBatchnorm within the code as well. this is not present in the current version

Thanks for taking the time in Fixing the issues. I will merge the PR.
